js/tji-datasets-explore.js

```diff
@@ -307,64 +309,124 @@ TJIChartView.prototype.transform_data = function() {
-  _.each(filters, function(f) {
-    var fieldset = jQuery('<fieldset><legend>' + f.key.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
-    _.each(f.values, function(v) {
-      var input = jQuery('<input/>', {
-        type: "checkbox",
-        checked: "checked",
-        id: v,
-        name: f.key,
-        value: v,
-      });
-      var label = jQuery('<label/>', {
-        for: v,
-      }).text(v);
-      fieldset.append(jQuery('<div></div>').append(input, label));
-    });
+  _.each(that.filter_configs, function(filter_config) {
+    var fieldset;
+    switch(filter_config.type) {
+      case 'autocomplete': 
+        fieldset = that.create_filter_autocomplete(filter_config);
+        break;
+      case 'checkbox':
+      default:
+        fieldset = that.create_filter_checkboxes(filter_config);
+        break;
+    }
     fieldsets.push(fieldset);
   });
   
-  this.$form = jQuery('<form />').append(fieldsets);
+  this.$form = jQuery('<form />', {
+    class: 'tji-chart-filters',
+  }).append(fieldsets);
   this.state.active_filters = this.$form.serializeArray();  
-  jQuery(this.filters_elt_id).append(this.$form);
+  jQuery(this.filters_elt_selector).append(this.$form);
   
 }
 
+TJIChartView.prototype.create_filter_checkboxes = function(filter) {
+  var that = this;
+  var fieldset = jQuery('<fieldset><legend>' + filter.name.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
+  _.each(filter.values, function(v) {
+    fieldset.append(that.create_filter_checkbox(filter.name, v));
+  });
+  return fieldset;
+}
+
+TJIChartView.prototype.create_filter_checkbox = function(name, value) {
+  var input = jQuery('<input/>', {
+      class: "tji-chart-filters__checkbox",
+      type: "checkbox",
+      checked: "checked",
+      id: 'TJIChartView__filter-' + value,
+      name: name,
+      value: value,
+    });
+  var label = jQuery('<label/>', {
+    for: 'TJIChartView__filter-' + value,
+  }).text(value);
+  var container = jQuery('<div/>', {
+      class: "tji-chart-filters__filter",
+  }).append(input, label);
+  return container;
+}
+
+TJIChartView.prototype.create_filter_autocomplete = function(filter) {
+  var that = this;
+  var fieldset = jQuery('<fieldset><legend>' + filter.name.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
+  var input = jQuery('<input/>', {
+    class: "tji-chart-filters__text",
+    type: "text",
+  });
+  fieldset.append(jQuery('<div/>',{
+    class: "tji-chart-filters__autocomplete",
+  }).append(input));
+  var $auto_complete = jQuery('<div />', {
+    class: "tji-chart-filters__autocomplete-list"
+  }).insertAfter(input);
+  var auto_complete = new autoComplete({
+      selector: input[0],
+      minChars: 1,
+      delay: 150, 
+      source: function(term, suggest){
+          term = term.toLowerCase();
+          suggest(_.filter(filter.values, function(v){
+            return ~v.toLowerCase().indexOf(term);
```

Comment:
For the love of god change this

![doit](https://user-images.githubusercontent.com/1421476/42423121-67d29f72-82ba-11e8-9875-08818cb68841.gif)

Reply:
never

Reply:
🙅‍♂️

Reply:
![nope](https://user-images.githubusercontent.com/3039125/42423849-e989a324-82c6-11e8-8722-54c8c1d4eaa2.gif)

Reply:
![justdoit](https://user-images.githubusercontent.com/1421476/42423866-5c0aa9e8-82c7-11e8-9449-382f42066e92.gif)